File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        int i,cnt,j;
        char x;
        char ch[] = s.toCharArray();
        for(i=0;i<s.length();i++)
        {
            cnt=0;
            x=ch[i];
            for(j=i;j<s.length()&&x!='^';j++)
            {
                if(x==ch[j])
                {
                    ch[j]='^';
                    cnt++;
                }
            }
            if(cnt==1)
            {
                return i;
            }
        }
        return -1;
    }
}

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map<int,int> x;
        for(int i=0;i<s.size();i++)
        {
            x[s[i]]++;
        }
        for(int i=0;i<s.size();i++)
        {
            if(x[s[i]]==1)
                return i;
        }
        return -1;
    }
};